main.py

```diff
@@ -0,0 +1,313 @@
+        # plt.show()
+
+
+def main():
+    VISUALIZE = True
+    REPLAY_RESULTS = False
+    LOAD_SAVED = REPLAY_RESULTS
+    LOGGING = False
+    GRAVITY = -9.81
+    if VISUALIZE:
+        p.connect(p.GUI)  # or p.DIRECT for nongraphical version
+    else:
+        p.connect(p.DIRECT)
+    p.setAdditionalSearchPath(pybullet_data.getDataPath())
+    p.setGravity(0, 0, GRAVITY)
+    planeId = p.loadURDF(Environment.plane_urdf_filepath,
+                         basePosition=[0, 0, -0.01])
+    kukaId = p.loadURDF(Environment.arm_filepath, basePosition=[0, 0, 0])
+    if LOGGING and VISUALIZE:
+        log_id = p.startStateLogging(
+            p.STATE_LOGGING_VIDEO_MP4, "avg_plan_success.mp4")
+
+    # bottle
+    # bottle_start_pos = np.array(
+    #     [-0, -0.6, Bottle.INIT_PLANE_OFFSET]).astype(float)
+    # bottle_goal_pos = np.array([-0.6, -0.2, 0]).astype(float)
+    bottle_start_pos = np.array(
+        [0.5, 0.5, 0]).astype(float)
+    bottle_goal_pos = np.array([0.2, 0.6, 0]).astype(float)
+    bottle_start_ori = np.array([0, 0, 0, 1]).astype(float)
+    bottle = Bottle(start_pos=bottle_start_pos, start_ori=bottle_start_ori)
+
+    # starting end-effector pos, not base pos
+    # NOTE: just temporarily setting arm to starting bottle position with some offset
+    # offset = -np.array([0.05, 0, 0])
+    # EE_start_pos = bottle_start_pos + offset
+    EE_start_pos = np.array([0.5, 0.3, 0.2])
+    base_start_ori = np.array([0, 0, 0, 1]).astype(float)
+    arm = Arm(EE_start_pos=EE_start_pos,
+              start_ori=base_start_ori,
+              kukaId=kukaId)
+    start_joints = arm.joint_pose
+
+    env = Environment(arm, bottle, is_viz=VISUALIZE)
+    start = np.concatenate(
+        [bottle_start_pos, start_joints])
+    # goal joints are arbitrary and populated later in planner
+    goal = np.concatenate(
+        [bottle_goal_pos, [0] * arm.num_joints])
+    xbounds = [-0.4, -0.9]
+    ybounds = [-0.1, -0.9]
+    dx = dy = dz = 0.1
+    dist_thresh = dx
+    # if  the below isn't true, you're expecting bottle to fall in exactly
+    # the same state bin as the goal
+    assert(dist_thresh <= dx)
+    eps = 5
+    da_rad = 8 * math.pi / 180.0
+
+    with open("filtered_start_goals.obj", "rb") as f:
+        start_goals = pickle.load(f)
+
+    # for i, (startb, goalb, start_joints) in enumerate(start_goals):
+    #     print(startb, goal)
+    #     if np.linalg.norm(np.array(startb) - np.array(goalb)) <= dist_thresh:
+    #         print("%d TOO EASY" % i)
+
+    # exit()
+
+    # change bottle with randomly sampled radius, height, fill, friction
+    # radius = 0.025
+    # height = 0.21
+    # fill_prop = 0.3
+    # fric = env.max_fric
+    # new_bottle = Bottle(start_pos=bottle_start_pos,
+    #                     start_ori=bottle_start_ori,
+    #                     fill_prop=fill_prop,
+    #                     fric=fric)
+    # env.bottle = new_bottle
+
+    # run planner and visualize result
+    num_sims_per_action = 20
```

Comment:
I find it convenient to centralize all my constants, so when I want to change one I don't have to go hunting around for it. I know in academia stuff is more ad-hoc so not really stating this as a problem.

But I do like the pattern you've established of naming the variable before using it, so you're clear on what the number is supposed to mean (vs just using it in-place).

Reply:
Oh yes, I will move all these constants to the top of the script so I know where to look next time. A lot of the bugs I saw in the past did involve adding hacky changes to these constants halfway in the code. I would see the initial definition of a constant, but the constant would unexpectedly change later on. By forcing everything to be defined at the top, this bug won't happen.